packages/flutter/lib/src/widgets/app.dart

```diff
@@ -108,6 +108,9 @@ typedef InitialRouteListFactory = List<Route<dynamic>> Function(String initialRo
 /// It is used by both [MaterialApp] and [CupertinoApp] to implement base
 /// functionality for an app.
 ///
+/// If a [MediaQuery] is available above this widget a new one will not be
+/// created by [WidgetsApp].
```

Comment:
Document that it builds a MediaQuery with data derived from WidgetsBinding.window if no MediaQuery is available?

Reply:
I eventually went with:
```dart
/// If a [MediaQuery] is not available above [WidgetsApp] a [MediaQuery] is
/// built using [MediaQueryFromWindow].
```
because it's concise and `builds a [MediaQuery] with data derived from [WidgetsBinding.window]` is already included under `See also: ... *[MediaQueryFromWindow]`.

I was previously considering:
```dart
/// Builds a [MediaQuery] with data derived from [WidgetsBinding.window] if
/// [MediaQuery] is not available above [WidgetsApp]. If a [MediaQuery] is 
/// available a new one will not be created by [WidgetsApp].
```

Let me know if you think that the second or some other variant would make more sense.

Reply:
This looks good, thanks for clarifying the docs.